src/products/index.ts

```diff
@@ -17,27 +18,52 @@ export type Transformer = (product: Product) => { data: object, variants: object
 @alias('products')
 @tag('gb-products', require('./index.html'))
 class Products {
-
   structure: Structure = this.config.structure;
+
+  props: Products.Props = {
+    storeSection: StoreSections.DEFAULT
+  };
+
   state: Products.State = {
     products: []
   };
   productTransformer: Transformer = ({ data, meta }: Product) =>
     ({ ...ProductTransformer.transformer(this.structure)(data), meta })
 
   init() {
-    this.flux.on(Events.PRODUCTS_UPDATED, this.updateProducts);
+    switch (this.props.storeSection) {
+      case StoreSections.PAST_PURCHASES:
+        this.flux.on(Events.PAST_PURCHASE_PRODUCTS_UPDATED, this.updateProducts);
+        break;
+      case StoreSections.DEFAULT:
+        this.flux.on(Events.PRODUCTS_UPDATED, this.updateProducts);
+    }
```

Comment:
if you are calling the same update function, why do you need a switch? To add cases in the future?

Reply:
They're called on different events though.